notebooks/20211225_Parent_HTML_Finder.ipynb

```diff
@@ -0,0 +1,809 @@
+    "scraped_pdfs = set(df_scraped_only['scraped_url'])\n",
+    "\n",
+    "pdfs_relevancy_mask = np.array(list(map(get_match_score, scraped_pdfs))).nonzero()\n",
+    "scraped_pdfs = np.array(list(scraped_pdfs))[pdfs_relevancy_mask]\n",
+    "\n",
+    "scraped_pdfs = sorted(scraped_pdfs, \n",
+    "                      key=get_match_score, reverse=True)\n",
+    "\n",
+    "print(f\"The following scraped PDFs were not stored by BankTrack, but match \"\n",
+    "      f\"keywords of interest and may be good to track.\\n\"\n",
+    "      f\"Sorted from most relevant to least relevant.\\n\")\n",
+    "for pdf in scraped_pdfs:\n",
+    "    print(pdf)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The following scraped HTMLS were not found when checking parent pages for PDFs stored by BankTrack, but match keywords of interest and may be good to track.\n",
+      "Sorted from most relevant to least relevant.\n",
+      "\n",
+      "https://www.citigroup.com//citi/sustainability/evolution.htm#Equator\n",
+      "https://www.citigroup.com//citi/sustainability/evolution.htm#green\n",
+      "https://www.citigroup.com/citi/about/citizenship/citi-impact-fund.html\n"
+     ]
+    }
+   ],
+   "source": [
+    "htmls_of_scraped_pdfs = \\\n",
+    "    set(df_all_links[df_all_links['scraped_url'].isin(scraped_pdfs)]['parent_htmls'].explode())\n",
+    "\n",
+    "scraped_htmls = set(df_scraped_only['parent_htmls'].explode())\n",
+    "scraped_htmls = scraped_htmls - match_parent_htmls\n",
+    "\n",
+    "scraped_htmls = sorted(scraped_htmls, \n",
+    "                       key=get_match_score, reverse=True)\n",
+    "\n",
+    "scraped_htmls = htmls_of_scraped_pdfs.union(scraped_htmls)\n",
+    "\n",
+    "htmls_relevancy_mask = np.array(list(map(get_match_score, scraped_htmls))).nonzero()\n",
+    "scraped_htmls = np.array(list(scraped_htmls))[htmls_relevancy_mask]\n",
+    "\n",
+    "scraped_htmls = set([html for html in scraped_htmls if 'news' not in html])\n",
+    "\n",
+    "print(f\"The following scraped HTMLS were not found when checking parent pages for PDFs stored by BankTrack, but match keywords of \"\n",
+    "      f\"interest and may be good to track.\\n\"\n",
+    "      f\"Sorted from most relevant to least relevant.\\n\")\n",
+    "for phtml in scraped_htmls - match_parent_htmls:\n",
+    "    print(phtml)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Observations\n",
+    "1. By a large mergin, the most links on BankTrack are stored internally and don't have an external link right now\n",
+    "2. Some pdfs can't be read. One reason is because sometimes the href is relative. Currently the script just prepends the base url, but this doesn't always work. We need to extract the url of the current page and prepend this.\n",
```

Comment:
I have also encountered this problem and I found some patterns to get more absolute working urls form the relative urls